x/pki/types/message_add_pki_revocation_distribution_point.go

```diff
@@ -0,0 +1,135 @@
+}
+
+func (msg *MsgAddPkiRevocationDistributionPoint) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Signer)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid signer address (%s)", err)
+	}
+
+	err = validator.Validate(msg)
+	if err != nil {
+		return err
+	}
+
+	allowedDataDigestTypes := [6]uint32{1, 7, 8, 10, 11, 12}
+	allowedRevocationType := uint64(1)
+
+	isDataDigestInTypes := false
+	for _, digestType := range allowedDataDigestTypes {
+		if digestType == msg.DataDigestType {
+			isDataDigestInTypes = true
+
+			break
+		}
+	}
+
+	if !isDataDigestInTypes {
+		return pkitypes.NewErrInvalidDataDigestType(fmt.Sprintf("invalid DataDigestType: %d", msg.DataDigestType))
+	}
+
+	if msg.RevocationType != allowedRevocationType {
+		return pkitypes.NewErrInvalidRevocationType(fmt.Sprintf("invalid RevocationType: %d", msg.RevocationType))
+	}
+
+	cert, err := x509.DecodeX509Certificate(msg.CrlSignerCertificate)
+	if err != nil {
+		return err
+	}
+
+	if msg.IsPAA {
+		if msg.Pid != 0 {
+			return pkitypes.NewErrNotEmptyPid("pid is not empty for root certificate")
+		}
+
+		if !cert.IsSelfSigned() {
+			return pkitypes.NewErrPAANotSelfSigned(fmt.Sprintf("root certificate with subject: %s, subjectKeyID: %s is not self-signed", cert.SubjectAsText, cert.SubjectKeyID))
+		}
+	} else {
+		if !strings.Contains(cert.SubjectAsText, string(msg.Pid)) {
+			return pkitypes.NewErrCRLSignerCertificateDoesNotContainPid(fmt.Sprintf("CRLSignerCertificate with subject: %s, subjectKeyID does not contain pid: %d", cert.SubjectAsText, cert.SubjectKeyID, msg.Pid))
+		}
+
+		if cert.IsSelfSigned() {
+			return pkitypes.NewErrNonPAASelfSigned(fmt.Sprintf("non-root certificate with subject: %s, subjectKeyID: %s is self-signed", cert.SubjectAsText, cert.SubjectKeyID))
+		}
+	}
+
+	if msg.DataFileSize == 0 && msg.DataDigest != "" {
+		return pkitypes.NewErrEmptyDataFileSize(fmt.Sprintf("msgAddRevocationDistributionPoint with CRLSignerCertificate: %s has empty DataFileSize when DataDigest is not empty", msg.CrlSignerCertificate))
+	}
+
+	if msg.DataFileSize != 0 && msg.DataDigest == "" {
+		return pkitypes.NewErrEmptyDataDigest(fmt.Sprintf("msgAddRevocationDistributionPoint with CRLSignerCertificate: %s has empty DataDigest when DataFileSize is not empty", msg.CrlSignerCertificate))
+	}
+
+	if msg.DataDigest == "" && msg.DataDigestType != 0 {
+		return pkitypes.NewErrEmptyDataDigest(fmt.Sprintf("msgAddRevocationDistributionPoint with CRLSignerCertificate: %s has empty DataDigest when DataDigestType is not empty", msg.CrlSignerCertificate))
+	}
+
+	if msg.DataDigest != "" && msg.DataDigestType == 0 {
+		return pkitypes.NewErrEmptyDataDigestType(fmt.Sprintf("msgAddRevocationDistributionPoint with CRLSignerCertificate: %s has empty DataDigestType when DataDigest is not empty", msg.CrlSignerCertificate))
+	}
+
+	if msg.RevocationType == 1 && (msg.DataFileSize != 0 || msg.DataDigest != "" || msg.DataDigestType != 0) {
```

Comment:
Please use a constant from `x/pki/types/types.go` instead of magic number `1` here and in other places.

Reply:
Fixed